--- 2008_0708_RT2860_Linux_STA_v1.7.0.0/os/linux/rt_main_dev.c  2008-06-3022:31:10.000000000 -0500
+++ 2008_0708_RT2860_Linux_STA_v1.7.0.0.new/os/linux/rt_main_dev.c     2008-09-05 12:53:04.000000000 -0500
@@ -801,7 +801,15 @@
            
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24)
-        device = dev_get_by_name(dev->nd_net, slot_name);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#ifdef CONFIG_NET_NS
+               device = dev_get_by_name(dev->nd_net, slot_name);
+#else
+               device = dev_get_by_name(&init_net, slot_name);
+#endif
+#else
+               device = dev_get_by_name(dev->nd_net, slot_name);
+#endif
 #else
                device = dev_get_by_name(slot_name);
 #endif
